pos robot_pos
pos[] obstacle_poses